server/fleet/integrations.go

```diff
@@ -15,120 +17,161 @@ type TeamIntegrations struct {
 	Zendesk []*TeamZendeskIntegration `json:"zendesk"`
 }
 
-// ToIntegrations converts a TeamIntegrations to an Integrations struct.
-func (ti TeamIntegrations) ToIntegrations() Integrations {
-	var intgs Integrations
-	intgs.Jira = make([]*JiraIntegration, len(ti.Jira))
-	for i, j := range ti.Jira {
-		intgs.Jira[i] = j.ToJiraIntegration()
+// MatchWithIntegrations matches the team integrations to their corresponding
+// global integrations found in globalIntgs, returning the resulting
+// integrations struct. It returns an error if any team integration does not
+// map to a global integration, but it will still return the complete list
+// of integrations that do match.
+func (ti TeamIntegrations) MatchWithIntegrations(globalIntgs Integrations) (Integrations, error) {
+	var result Integrations
+
+	jiraIntgs, err := IndexJiraIntegrations(globalIntgs.Jira)
+	if err != nil {
+		return result, err
+	}
+	zendeskIntgs, err := IndexZendeskIntegrations(globalIntgs.Zendesk)
+	if err != nil {
+		return result, err
+	}
+
+	var errs strings.Builder
+	for _, tmJira := range ti.Jira {
+		key := tmJira.UniqueKey()
+		intg, ok := jiraIntgs[key]
+		if !ok {
+			if errs.Len() > 0 {
+				errs.WriteByte('\n')
+			}
+			fmt.Fprintf(&errs, "unknown Jira integration for url %s and project key %s", tmJira.URL, tmJira.ProjectKey)
```

Comment:
just curious: did you decide to use a `strings.Builder` for performance? Seems like the easiest way to go would be a `errs := []string{}` and then `strings.Join(errs, "\n")`

Reply:
No specific reason other than generally being the best type to build that, but it's true that it'd make the code a bit nicer with the slice and join, I'll go with that!